src/IO/S3/copyS3File.cpp

```diff
@@ -746,7 +746,10 @@ namespace
                     break;
                 }
 
-                if (outcome.GetError().GetExceptionName() == "EntityTooLarge" || outcome.GetError().GetExceptionName() == "InvalidRequest" || outcome.GetError().GetExceptionName() == "InvalidArgument")
+                if (outcome.GetError().GetExceptionName() == "EntityTooLarge" ||
+                    outcome.GetError().GetExceptionName() == "InvalidRequest" ||
+                    outcome.GetError().GetExceptionName() == "InvalidArgument" ||
+                    (outcome.GetError().GetExceptionName() == "InternalError" && outcome.GetError().GetResponseCode() == Aws::Http::HttpResponseCode::GATEWAY_TIMEOUT))
```

Comment:
Checking for `GATEWAY_TIMEOUT` doesn't seem completely correct because it can mean something else.
Can we check something else to identify that kind of response?

Reply:
For example, we could search in the response for this text "use the Rewrite method in the JSON API` instead.

Reply:
Agree. I think we need to change this check a bit.
But I am not sure if check for some specific substring in error message is okay.
@antonio2368 What do you think ?

Reply:
hmm I agree with @vitlibar, we would risk fallback to heavy operation instead of simply retrying the copy if the error can be avoided by a simple retry.
and both the code and exception name seem generic so we would have to search in the respones

Reply:
Updated.